### DISCUSSION TITLE
Midi keyboard style control, importing or converting styles

### DISCUSSION THREAD

**Discussion by jenicek-sut** [ensemblesaw/ensembles-app#88] [Ideas]
Hello. This is a program I've been looking for for a long time for Linux. Looks great. Would it be possible to add the option to use yamaha styles, or a conversion utility? Furthermore, and I think this is very necessary - control of the automatic accompaniment during the game using the midi keyboard (change of variation, ending, tempo, etc.) Thank you for your work. Jan

**Comment by SubhadeepJasu** [comment]
Thanks! I am glad you like it. 

Midi style control using keyboard already works. You can right click the style control buttons and assign a control by moving it on your keyboard. In my case I just use the first couple of keys on my keyboard. (at the moment it doesn't remember the binding which I am working on a fix)

Supporting yamaha styles or creating a style conversion utility is not possible due to legal reasons.

**Reply by jenicek-sut** [reply]
Thank you. I tried assigning the control using the keyboard as follows (failure): I hold down the sync start button with the right mouse and simultaneously press the key I want to assign - eg C4. I tried other combinations. It doesn't work.

**Reply by SubhadeepJasu** [reply]
If you right click on the sync button, do you get this?
![Screenshot from 2023-11-26 18 11 15](https://github.com/ensemblesaw/ensembles-app/assets/20795161/8449f261-a34c-4287-a48e-bfd156485b33)

Clicking that does this:
![Screenshot from 2023-11-26 18 12 18](https://github.com/ensemblesaw/ensembles-app/assets/20795161/ff0f4dea-fdfd-4c07-9df3-870d42fe14fd)
